### PR TITLE
Forget password

### LINKED ISSUE #109: Forget password API
**Is your feature request related to a problem? Please describe.**
as a user i should be have the option of forget password.

**Describe the solution you'd like**
1. create [post] `api/users/forget_password/` which takes in the email of the user and send an OTP to the registered email.
1. create `otp` and `otp_created_at` field in the user model.

**Additional context**
1. OTP should be valid for 1hr. 
1. if user requests OTP within 1hr mark then send the same OTP else create a new OTP.
1. OTP should be a 5 digit integer.

### PR DESCRIPTION
**Pull request by anshsrtv**
### Description
The forget password API is ready and tested with relevant testcases

**Type of Change:** (Delete irrelevant options)
- Code
- Testing
- Documentation

**Code/Quality Assurance Only** (Delete irrelevant options)

- Bug fix (`MOCK_EMAIL = True` and printing body in mock email)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
This has been tested using the django test cases and using swagger docs.
```
ansh@ansh-HP-Notebook:~/Desktop/teesco-backend$ python manage.py test 
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
..........................................................
----------------------------------------------------------------------
Ran 58 tests in 18.130s

OK
Destroying test database for alias 'default'...
```

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged
- [x] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: users/views.py [hunk 1]
```diff
@@ -75,7 +78,76 @@ def post(self, request):
...
+                one_hour = datetime.timedelta(hours = 1)
+
+                # Mail Variables
+                subject = 'OTP to Reset your Password'
+                body = "Dear user,</br></br>\
+                <b>The OTP to reset your password is {}</b>.</br>\
+                Please do not share it with anyone.</br>\
+                </br>\
+                Best Regards,</br>\
+                E-Cell NITRR Open Source"
```
Comment (naveennvrgup) [inline users/views.py]:
```suggestion
                Teesco (Volunteer Management System)"
```

**Review by naveennvrgup — CHANGES_REQUESTED** [review]
1. Please revert changes made to `org/responses.py`
1. Great quality code. appreciate the efforts you made for documentation :clap: :medal_military:

**Comment by naveennvrgup** [comment]
Tested locally works great :+1:

**Comment by anshsrtv** [comment]
> 1. Please revert changes made to `org/responses.py`
> 2. Great quality code. appreciate the efforts you made for documentation clap medal_military

Made the changes :v:

**Review by priyanshisharma — APPROVED** [review]
Tested it locally, Looks right! 🎉

File: utils/email_service.py [hunk 2]
```diff
@@ -1,51 +1,51 @@
...
+                raise ValueError("Invalid AWS credentials!")
+
+        elif settings.EMAIL_SERVICE == 'DJANGO-SMTP':
+            try:
+                msg = EmailMessage(subject, body, settings.EMAIL_HOST_USER, recipient)
+                msg.content_subtype = "html"
+
+                msg.send()
+            except:
+                raise ValueError("Invalid Gmail credentials!")
```
Comment (sahil9001) [inline utils/email_service.py]: `email_service.py` was already added in the previous PR. :sweat_smile:

**Review by sahil9001 — CHANGES_REQUESTED** [review]

**Review by naveennvrgup — APPROVED** [review]
1. Tested locally works as expected.
1. Great quality code :clap:

**Pull request merged**